tidy3d/components/geometry.py

```diff
@@ -3281,8 +3290,269 @@ def _surface_area(self, bounds: Bound) -> float:
+            within a given triangle, and the third index is the coordinate (x, y, or z).
+
+        Returns
+        -------
+        :class:`.CustomSurfaceMeshGeometry`
+            The custom surface mesh geometry given by the triangles provided.
+
+        """
+        num_facets = len(triangles)
+        coords = dict(
+            facet_index=np.arange(num_facets),
+            vertex_index=np.arange(3),
+            axis=np.arange(3),
+        )
+        vertices = SurfaceMeshDataArray(triangles, coords=coords)
+        mesh_dataset = SurfaceMeshDataset(triangles=vertices)
+        return CustomSurfaceMeshGeometry(mesh_dataset=mesh_dataset)
+
+    @classmethod
+    def from_vertices_faces(
+        cls, vertices: np.ndarray, faces: np.ndarray
+    ) -> CustomSurfaceMeshGeometry:
+        """Create a :class:`.CustomSurfaceMeshGeometry` from numpy arrays containing the data
+        of a surface mesh. The first array contains the vertices, and the second array contains
+        faces formed from triples of the vertices.
+
+        Parameters
+        ----------
+        vertices: ``np.ndarray``
+            A numpy array of shape (N, 3) storing the vertices of the surface mesh.
+            The first index labels the vertex, and the second index is the coordinate
+            (x, y, or z).
+        faces : ``np.ndarray``
+            A numpy array of shape (M, 3) storing the indices of the vertices of each face
+            in the surface mesh. The first index labels the face, and the second index
+            labels the vertex index within the ``vertices`` array.
+
+        Returns
+        -------
+        :class:`.CustomSurfaceMeshGeometry`
+            The custom surface mesh geometry given by the vertices and faces provided.
+
+        """
+        return cls.from_triangles(trimesh.Trimesh(vertices, faces).triangles)
+
+    @classmethod
+    def _triangles_to_trimesh(cls, triangles: np.ndarray) -> trimesh.Trimesh:
+        """Convert an (N, 3, 3) numpy array of triangles to a ``trimesh.Trimesh``."""
+        return trimesh.Trimesh(**trimesh.triangles.to_kwargs(triangles))
+
+    @cached_property
+    def trimesh(self) -> trimesh.Trimesh:
+        """A ``trimesh.Trimesh`` object representing the custom surface mesh geometry."""
+        return self._triangles_to_trimesh(self.triangles)
+
+    @cached_property
+    def triangles(self) -> np.ndarray:
+        """The triangles of the surface mesh as an ``np.ndarray``."""
+        if self.mesh_dataset is None:
+            raise DataError("Can't get triangles as 'mesh_dataset' is None.")
+        return self.mesh_dataset.triangles.to_numpy()
+
+    def _surface_area(self, bounds: Bound) -> float:
+        """Returns object's surface area given bounds."""
+        # currently ignores bounds
+        return self.trimesh.area
+
+    def _volume(self, bounds: Bound) -> float:
+        """Returns object's volume given bounds."""
+        # currently ignores bounds
```

Comment:
sorry, I think at some point the docstrings used regular comments `#` instead of `"""`
like this
```py
def _volume(...):
    # docstring
```

maybe I was mistaken